pkg/server/datastore/sqlstore/sqlstore_test.go

```diff
@@ -1359,30 +1363,60 @@ func (s *PluginSuite) TestPruneRegistrationEntries() {
 
 	createdRegistrationEntry, err := s.ds.CreateRegistrationEntry(ctx, entry)
 	s.Require().NoError(err)
+	fetchedRegistrationEntry := &common.RegistrationEntry{}
+	defaultLastLog := spiretest.LogEntry{
+		Level:   logrus.InfoLevel,
+		Message: "Connected to SQL database",
+		Data: logrus.Fields{
+			telemetry.Type:     "sqlite3",
+			telemetry.Version:  "3.36.0",
+			telemetry.ReadOnly: "false",
+		},
+	}
 
-	// Ensure we don't prune valid entries, wind clock back 10s
-	err = s.ds.PruneRegistrationEntries(ctx, now.Add(-10*time.Second))
-	s.Require().NoError(err)
-
-	fetchedRegistrationEntry, err := s.ds.FetchRegistrationEntry(ctx, createdRegistrationEntry.EntryId)
-	s.Require().NoError(err)
-	s.Equal(createdRegistrationEntry, fetchedRegistrationEntry)
-
-	// Ensure we don't prune on the exact ExpiresBefore
-	err = s.ds.PruneRegistrationEntries(ctx, now)
-	s.Require().NoError(err)
-
-	fetchedRegistrationEntry, err = s.ds.FetchRegistrationEntry(ctx, createdRegistrationEntry.EntryId)
-	s.Require().NoError(err)
-	s.Equal(createdRegistrationEntry, fetchedRegistrationEntry)
-
-	// Ensure we prune old entries
-	err = s.ds.PruneRegistrationEntries(ctx, now.Add(10*time.Second))
-	s.Require().NoError(err)
+	for _, tt := range []struct {
+		name                      string
+		time                      time.Time
+		expectedRegistrationEntry *common.RegistrationEntry
+		expectedLastLog           spiretest.LogEntry
+	}{
+		{
+			name:                      "Don't prune valid entries",
+			time:                      now.Add(-10 * time.Second),
+			expectedRegistrationEntry: createdRegistrationEntry,
+			expectedLastLog:           defaultLastLog,
+		},
+		{
+			name:                      "Don't prune exact ExpiresBefore",
+			time:                      now,
+			expectedRegistrationEntry: createdRegistrationEntry,
+			expectedLastLog:           defaultLastLog,
+		},
+		{
+			name:                      "Prune old entries",
+			time:                      now.Add(10 * time.Second),
+			expectedRegistrationEntry: (*common.RegistrationEntry)(nil),
+			expectedLastLog: spiretest.LogEntry{
+				Level:   logrus.InfoLevel,
+				Message: "Pruned an expired registration",
+				Data: logrus.Fields{
+					telemetry.SPIFFEID: entry.SpiffeId,
+					telemetry.ParentID: entry.ParentId,
+				},
+			},
+		},
+	} {
+		tt := tt
+		s.T().Run(tt.name, func(t *testing.T) {
+			err = s.ds.PruneRegistrationEntries(ctx, tt.time)
+			s.Require().NoError(err)
+			fetchedRegistrationEntry, err = s.ds.FetchRegistrationEntry(ctx, createdRegistrationEntry.EntryId)
+			s.Require().NoError(err)
+			s.Equal(tt.expectedRegistrationEntry, fetchedRegistrationEntry)
```

Comment:
We should not use the Suite functions to do the asserts within the subtest or the testing package will complain. Instead the `*testing.T` passed into the subtest should be used with the functions in the `require` or `assert` packages (e.g. `require.NoError(t, err)`).

Reply:
Got it, thanks!